### PR TITLE
Adding Broker read permissions

### PR DESCRIPTION
**Pull request by nachocano**
## Proposed Changes

  * Adding read only permissions to brokers.
  * This should be merged right after https://github.com/knative/eventing/pull/788, otherwise pointing sources to brokers won't work. 

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: config/201-clusterrole.yaml [hunk 1]
```diff
@@ -86,3 +86,10 @@ rules:
   resources:
   - events
   verbs: *everything
+
+# Brokers read
```
Comment (evankanderson) [inline config/201-clusterrole.yaml]: Put this next to Channels read?

Comment (nachocano) [reply config/201-clusterrole.yaml]: you mean this?

```
# Channels and Brokers read
- apiGroups:
  - eventing.knative.dev
  resources:
  - channels
  - brokers
  verbs: &readOnly
  - get
  - list
  - watch
```

or just as a different block below channels?

Comment (nachocano) [reply config/201-clusterrole.yaml]: Done

**Comment by nachocano** [comment]
> Do you also need to update:
> 
> https://github.com/knative/eventing-sources/blob/master/contrib/gcppubsub/config/201-clusterrole.yaml
> https://github.com/knative/eventing-sources/blob/master/contrib/camel/config/201-clusterrole.yaml

Thanks! Done

File: config/201-clusterrole.yaml [hunk 2]
```diff
@@ -56,11 +56,12 @@ rules:
   - deployments
   verbs: *everything
 
-# Channels read
+# Channels and Brokers read
```
Comment (matzew) [inline config/201-clusterrole.yaml]: :+1:

**Comment by Harwayne** [comment]
This might be a good case for [Aggregated ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles). 

Essentially, we label `ClusterRole`s that are needed to for resolving Addressables, then all `ServiceAccount`s that need to resolve Addressables simply get the aggregated role. Then any updates just need to be made in one place and suddenly all `ServiceAccount`s have the correct permissions.

**Comment by Harwayne** [comment]
> This might be a good case for [Aggregated ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles).
> 
> Essentially, we label `ClusterRole`s that are needed to for resolving Addressables, then all `ServiceAccount`s that need to resolve Addressables simply get the aggregated role. Then any updates just need to be made in one place and suddenly all `ServiceAccount`s have the correct permissions.

Created https://github.com/knative/eventing/issues/916 to track this idea.

**Comment by evankanderson** [comment]
You need to remove the "[WIP]" from the title when you want Prow to merge this.

**Comment by nachocano** [comment]
> You need to remove the "[WIP]" from the title when you want Prow to merge this.

Done! Thanks @evankanderson, my bad.

**Pull request merged**